libs/wingc/src/type_check.rs

```diff
@@ -194,23 +211,23 @@ impl PartialEq for Type {
 				// If our parent is equal to `other` then treat both classes as equal (inheritance)
 				if let Some(parent) = l0.parent.as_ref() {
 					let parent_type: &Type = &*parent;
-					return parent_type.eq(other);
+					return parent_type.is_subtype_of(other);
 				}
 				false
 			}
 			(Self::Resource(l0), Self::Resource(_)) => {
 				// If our parent is equal to `other` then treat both resources as equal (inheritance)
 				if let Some(parent) = l0.parent.as_ref() {
 					let parent_type: &Type = &*parent;
-					return parent_type.eq(other);
+					return parent_type.is_subtype_of(other);
 				}
 				false
 			}
 			(Self::Struct(l0), Self::Struct(_)) => {
 				// If we extend from `other` then treat both structs as equal (inheritance)
```

Comment:
Need to remove the usage of "equal" from all these comments.
```suggestion
				// If we extend from `other` then I'm a subtype of it (inheritance)
```